File: Python/leetcode_1365.py
from typing import List


def smaller_numbers_than_current(nums: List[int]) -> List[int]:
    original = nums[:]
    nums.sort()
    nums.reverse()
    m = {}
    ln = len(nums)
    cn = 0
    for num in nums:
        cn += 1
        m[num] = ln - cn
    return [m[x] for x in original]


print(smaller_numbers_than_current([7, 7, 7, 7]))

File: JavaScript/leetcode_1365.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallerNumbersThanCurrent = function (nums) {
    const original = [...nums];
    nums.sort((a, b) => a - b);
    let m = new Map();
    for (let index = 0; index < nums.length; index++) {
        if (!m.has(nums[index])) m.set(nums[index], index);
    }
    return original.map((n) => m.get(n));
};

console.log(smallerNumbersThanCurrent([8, 1, 2, 2, 3]));
